### PR TITLE
Release/v0.31.1

### PR DESCRIPTION
**Pull request by Liamsi**
https://github.com/tendermint/tendermint/blob/ae88965ff6ddf174607a02eee83cfc5ed11c6cbb/CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,53 @@
...
+- [mempool] [\#2778](https://github.com/tendermint/tendermint/issues/2778) No longer send txs back to peers who sent it to you and limit to 65536 active peers.
+This vastly improves the the b bandwidth consumption of nodes. 
+E.g. 250bytes txs for 120 sec. at 500 txs/sec. rate, so total 15MB:
+  - total bytes received from 1st node in 4 node localnet
+     - before: 42793967 (43MB)
+     - after: 30003256 (30MB)                                                                         
+  - total bytes sent to 1st node in 4 node localnet
+     - before: 30569339 (30MB)
+     - after: 19304964 (19MB)  
+- [p2p] [\#3475](https://github.com/tendermint/tendermint/issues/3475) Simplify `GetSelectionWithBias` for addressbook
```
Comment (melekes) [inline CHANGELOG.md]:
```suggestion
 - [p2p] [\#3475](https://github.com/tendermint/tendermint/issues/3475) Simplify `GetSelectionWithBias` for addressbook (@guagualvcha)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,53 @@
 # Changelog
 
+## v0.31.1
+
+*March 27th, 2019*
+
```
Comment (melekes) [inline CHANGELOG.md]: Don't you want to write short description for the release?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,53 @@
...
+### BREAKING CHANGES:
+
+* CLI/RPC/Config
+
+* Apps
+
+* Go API
+  - [crypto] [\#3426](https://github.com/tendermint/tendermint/pull/3426) Remove `Ripemd160` helper method (@needkane)
+  - [libs/common] Remove `RepeatTimer` (also `TimerMaker` and `Ticker` interface)
```
Comment (Liamsi) [inline CHANGELOG.md]: Forgot to link the issue / PR here.

File: CHANGELOG.md (hunk 2 above)
Comment (Liamsi) [reply CHANGELOG.md]: Is this OK:
This release contains improvements for the mempool that reduce the bandwidth requirements for nodes 
(some numbers are provided below).   
We also include a fix to a memory leak in the mempool and RPC HTTP servers can now be run over TLS.
Besides that it contains minor improvements, bugfixes, and code cleanups.

Comment (melekes) [reply CHANGELOG.md]:
```
This release contains a major improvement for the mempool that reduces the amount of data sent by ~30% (see the numbers below). It also fixes a memory leak in the mempool and gives you ability to run HTTPs (over TLS) RPC server by providing certificate and key in the config.
```
I'd remove `Besides that it contains minor improvements, bugfixes, and code cleanups.`

**Comment by Liamsi** [comment]
@melekes  https://github.com/tendermint/tendermint/pull/3490

**Comment by ebuchman** [comment]
Seems to be a race condition:

```
==================
WARNING: DATA RACE
Read at 0x00c000152743 by goroutine 52:
  testing.(*common).logDepth()
      /usr/local/go/src/testing/testing.go:621 +0x94
  testing.(*common).Logf()
      /usr/local/go/src/testing/testing.go:614 +0x90
  github.com/tendermint/tendermint/blockchain.testPeer.simulateInput()
      /go/src/github.com/tendermint/tendermint/blockchain/pool_test.go:45 +0x1a2
  github.com/tendermint/tendermint/blockchain.testPeer.runInputRoutine.func1()
      /go/src/github.com/tendermint/tendermint/blockchain/pool_test.go:36 +0x123

Previous write at 0x00c000152743 by goroutine 48:
  testing.tRunner.func1()
      /usr/local/go/src/testing/testing.go:856 +0x33e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:869 +0x17f

Goroutine 52 (running) created at:
  github.com/tendermint/tendermint/blockchain.testPeer.runInputRoutine()
      /go/src/github.com/tendermint/tendermint/blockchain/pool_test.go:34 +0x6a
  github.com/tendermint/tendermint/blockchain.testPeers.start()
      /go/src/github.com/tendermint/tendermint/blockchain/pool_test.go:52 +0x10c
  github.com/tendermint/tendermint/blockchain.TestBlockPoolBasic()
      /go/src/github.com/tendermint/tendermint/blockchain/pool_test.go:87 +0x1ef
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163

Goroutine 48 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:916 +0x699
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1157 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:865 +0x163
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1155 +0x523
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1072 +0x2eb
  github.com/tendermint/tendermint/blockchain.TestMain()
      /go/src/github.com/tendermint/tendermint/blockchain/store_test.go:111 +0x5cb
  main.main()
      _testmain.go:114 +0x330
==================

```

**Comment by Liamsi** [comment]
regarding the race condition, ref: https://github.com/tendermint/tendermint/issues/3390#issue-418379890

**Comment by Liamsi** [comment]
@melekes restarted the build. Regarding the race-condition: it does not seem critical and https://github.com/tendermint/tendermint/pull/3500#issuecomment-477545849 "fixes" it. We should not include this PR in this release tho.

**Pull request merged**